src/bun.js/webcore/response.zig

```diff
@@ -2141,24 +2142,55 @@ pub const Fetch = struct {
             const PercentEncoding = @import("../../url.zig").PercentEncoding;
             var path_buf2: [bun.MAX_PATH_BYTES]u8 = undefined;
             var stream = std.io.fixedBufferStream(&path_buf2);
-            const url_path_decoded = path_buf2[0 .. PercentEncoding.decode(
+            var url_path_decoded = path_buf2[0 .. PercentEncoding.decode(
                 @TypeOf(&stream.writer()),
                 &stream.writer(),
                 url.path,
-            ) catch {
-                globalThis.throwOutOfMemory();
+            ) catch |err| {
+                globalThis.throwError(err, "Failed to decode file url");
                 return .zero;
             }];
-            const temp_file_path = bun.path.joinAbsStringBuf(
-                globalThis.bunVM().bundler.fs.top_level_dir,
-                &path_buf,
-                &[_]string{
-                    globalThis.bunVM().main,
-                    "../",
-                    url_path_decoded,
-                },
-                .auto,
-            );
+
+            const temp_file_path = brk: {
+                if (std.fs.path.isAbsolute(url_path_decoded)) {
+                    if (Environment.isWindows) {
+                        // pathname will start with / if is a absolute path on windows, so we remove before normalizing it
+                        if (url_path_decoded[0] == '/') {
+                            url_path_decoded = url_path_decoded[1..];
+                        }
+                        break :brk PosixToWinNormalizer.resolveCWDWithExternalBufZ(&path_buf, url_path_decoded) catch |err| {
+                            globalThis.throwError(err, "Failed to resolve file url");
+                            return .zero;
+                        };
+                    }
+                    break :brk url_path_decoded;
+                }
+
+                var cwd_buf: [bun.MAX_PATH_BYTES]u8 = undefined;
+                const cwd = std.os.getcwd(&cwd_buf) catch |err| {
```

Comment:
Is this necessary on posix?

Reply:
Fix here: https://github.com/oven-sh/bun/pull/8784/commits/4fd215724bfae7389d190aa041a050c592f77bd2